src/vs/platform/files/browser/indexedDBFileSystemProvider.ts

```diff
@@ -426,8 +427,16 @@ export class IndexedDBFileSystemProvider extends Disposable implements IFileSyst
 	private fileWriteBatch: { resource: URI; content: Uint8Array }[] = [];
 	private async writeMany() {
 		if (this.fileWriteBatch.length) {
+			const estimated = await navigator?.storage?.estimate?.();
```

Comment:
Is it needed to be computed always? Or can this be computed just once and cached?

Reply:
I think it has to be recomputed always. This is because the estimated storage quota/usage is tracked for a group of origins, e.g. insiders.vscode.dev and vscode.dev share the same origin, so this isn't a quota that the IndexedDB filesystem provider is completely in control of. I tried to lessen the impact of this by only calling the storage API once per write batch rather than for every write.

Reply:
I am wondering if we can skip checking it upfront and check it when the error is thrown? I do not want to pay this penality for the majoirty of cases.